mempool/clist_mempool.go

```diff
@@ -163,29 +164,28 @@ func (mem *CListMempool) CloseWAL() {
 	mem.wal = nil
 }
 
-// Safe for concurrent use by multiple goroutines.
-func (mem *CListMempool) Lock() {
-	mem.updateMtx.Lock()
-}
-
-// Safe for concurrent use by multiple goroutines.
-func (mem *CListMempool) Unlock() {
-	mem.updateMtx.Unlock()
+// Meta returns the mempool metadata, including size and TXsBytes.
+func (mem *CListMempool) Meta() Meta {
+	return Meta{
+		Size:     mem.txs.Len(),
+		TXsBytes: atomic.LoadInt64(&mem.txsBytes),
+	}
 }
 
-// Safe for concurrent use by multiple goroutines.
-func (mem *CListMempool) Size() int {
-	return mem.txs.Len()
-}
+// PrepBlockFinality prepares the mempool for an upcoming block to be finalized. During
+// the execution of the new block, we want to make sure we are not running any additional
+// CheckTX calls to the application as well as flush any active connections underway.
+func (mem *CListMempool) PrepBlockFinality(ctx context.Context) (func(), error) {
```

Comment:
previously, we had exposed the actual Lock/Unlock and FlushSync methods for block executor to wield. It feels like we are leaking a lot of implementation details. All we're doing with those methods, is preparing the mempool for a block we want to finalize. This just wraps that up with a more obvious name. This also gives the mempool some freedom to
prepare in different ways. Not all mempools are created equal and require the locking.